common/tools/dev-tool/src/util/resolveTsConfig.ts

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import * as path from "path";
+import * as fs from "fs";
+
+function resolveConfigDir(configPath: string, unresolvedPath: string) {
+  return unresolvedPath.replace(/\$\{configDir\}/g, path.dirname(configPath));
+}
+
+/**
+ * Generates a complete TypeScript configuration by reading and parsing a given tsconfig file.
+ *
+ * @param configPath - The path to the tsconfig file.
+ * @returns A record containing the raw parsed configuration.
+ */
+export function resolveConfig(configPath: string) {
+  function helper(configPath: string) {
+    const absolutePath = path.resolve(configPath);
+    const configDir = path.dirname(absolutePath);
+    const rawConfig = JSON.parse(
+      fs.readFileSync(
+        !absolutePath.toLowerCase().endsWith(".json") ? `${absolutePath}.json` : absolutePath,
+        "utf8",
+      ),
+    );
+
+    let resolvedConfig = rawConfig;
```

Comment:
Does the ordering of base configs in `rawConfig.extends` matter? Should we start with an empty object and merge each config into it?  The loop below may have the issue of giving `base1.option1` and `base2.option1`, in first iteration `resolvedConfig` gets `base1.option1`, then in second iteration it overrides `base2.option1` with `base1.option1`.

or maybe do the merge in reverse order?

Reply:
Ordering matters, later items in the array have higher precedence. This mirrors behavior of the --showConfig flag.

Reply:
It is fixed with added tests in https://github.com/Azure/azure-sdk-for-js/pull/32087/commits/10a4d51c5ac4b8aaaccb4421710240d362b2ad2b